spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -881,21 +885,42 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
       raise "Action not provided for this tester type." unless url
 
       tester_data = {
-            emailAddress: {
-              value: email
-            },
-            firstName: {
-              value: first_name || ""
-            },
-            lastName: {
-              value: last_name || ""
-            },
-            testing: {
-              value: true
+        emailAddress: {
+          value: email
+        },
+        firstName: {
+          value: first_name || ""
+        },
+        lastName: {
+          value: last_name || ""
+        },
+        testing: {
+          value: true
+        }
+      }
+
+      if groups
+        tester_data[:groups] = groups.map do |group_name_or_group_id|
+          if self.groups.values.include?(group_name_or_group_id)
+            # This is an existing group, let's use that, the user specified the group name
+            group_name = group_name_or_group_id
+            group_id = self.groups.find { |id, name| name == group_name_or_group_id }[0]
+          elsif self.groups.keys.include?(group_name_or_group_id)
+            # This is an existing group, let's use that, the user specified the group ID
+            group_name = self.groups.find { |id, name| id == group_name_or_group_id }[1]
+            group_id = group_name_or_group_id
+          else
+            group_name = group_name_or_group_id
+            group_id = nil # this is expected by the iTC API
+          end
+
+          {
+            "id" => group_id,
+            "name" => {
+              "value" => group_name
```

Comment:
Not a big fan of adding a new method and clutter the class' methods scope for this, as we won't need to be calling it from anywhere else. Is there any better way?